cql2pgjson/src/main/java/org/folio/cql2pgjson/CQL2PgJSON.java

```diff
@@ -542,7 +542,9 @@ private String subQuery(String index,CQLTermNode node, Table correlation) throws
     String[] termParts = node.getTerm().split("\\.");
 
     String [] foreignTarget = (idxParts.length > termParts.length) ? idxParts : termParts;
-    ForeignKeys fkey = findForeignKey(dbTable.getPkColumnName(),correlation);
+    boolean indexInTable = findForeignKey(dbTable.getPkColumnName(),correlation, dbTable) == null;
+    ForeignKeys fkey = indexInTable ? findForeignKey(PK_COLUMN_NAME,dbTable, correlation) : findForeignKey(dbTable.getPkColumnName(),correlation, dbTable);
```

Comment:
This is called two times in the two lines, can this be avoided?

    findForeignKey(dbTable.getPkColumnName(),correlation, dbTable)

Reply:
Can you add a space after each comma?